### ISSUE TITLE
Presentation breaks with Error: Maximum update depth exceeded. 

### ISSUE THREAD

**Issue by mdillgen** [sanity-io/visual-editing#1486]
**Describe the bug**
Latest update breaks presentation and throws an error immediately!

**To Reproduce**
https://github.com/sanity-io/visual-editing/issues/1486#issuecomment-2103758981   ⬇️

**Which versions of Sanity are you using?**

```
@sanity/cli (global)                   3.41.1 (up to date)
@sanity/document-internationalization   3.0.0 (up to date)
@sanity/image-url                       1.0.2 (up to date)
@sanity/preview-url-secret             1.6.12 (up to date)
@sanity/react-loader                   1.9.17 (up to date)
@sanity/types                          3.41.1 (up to date)
@sanity/ui                              2.1.6 (up to date)
@sanity/vision                         3.41.1 (up to date)
@sanity/visual-editing                 1.8.18 (up to date)
sanity                                 3.41.1 (up to date)
```

**What operating system are you using?**
macOS 14.4.1 

**Which versions of Node.js / npm are you running?**

Run `npm -v && node -v` in the terminal and copy-paste the result here.
```
npm 10.1.0
Node v20.9.0
```

**Comment by ryanleichty** [comment]
I'm running into the same issue with the latest version.

![cleanshot_2024-05-07_at_21 30 40_2x](https://github.com/sanity-io/visual-editing/assets/17421429/6b5121e6-875a-4c20-9290-6bab4a98b973)

**Comment by christianhg** [comment]
Thank you for reporting this. Can you elaborate a bit on your reproduction steps. In what repository did you reproduce it?

**Comment by mdillgen** [comment]
> Thank you for reporting this. Can you elaborate a bit on your reproduction steps. In what repository did you reproduce it?

Hey @christianhg ,

Thank you for your reply, and sorry for the lack of detail in **the reproduction steps:**

1️⃣ To be fair, I set up a fresh repo, forked from the great template-nextjs-personal-website:
[https://github.com/mdillgen/template-nextjs-personal-website](https://github.com/mdillgen/template-nextjs-personal-website)

2️⃣ Updated dependencies
❌ Error occurs!

_Note that I also fixed the deprecated locate in the config, as you can see in the latest commit. BUT the error still happens._

**Comment by ryanleichty** [comment]
@christianhg I'm also using the `template-nextjs-personal-website` template for the site I'm working on. I just created another fresh repo using it to see if the changes I've made had anything to do with the error, and the same issue happens as soon as I update Sanity. You can see it here: https://github.com/ryanleichty/nextjs-personal-website

**Comment by christianhg** [comment]
> 2️⃣ Updated dependencies

> issue happens as soon as I update Sanity

Thanks, that makes sense. For now, I would suggest pinning `sanity` to `3.40.0` until we've found a fix for this issue.

**Comment by christianhg** [comment]
A fix for this problem has been implemented in #1496 and released as part of version `presentation@1.15.2`. Hopefully, you'll find the fix is working for you as well as soon as the latest version lands in your `node_modules`.
